python/util/templates.py

```diff
@@ -59,22 +60,22 @@ def compile_component(self, component_name):
         component = self[component_name]
         return reactjs.component_to_javascript(component.name, component.body)
 
-    def generate_react_components(self):
+    def generate_react_components(self, user_embed, job_embed, domain_embed):
         """ Generate javascript to render every component which has been marked
             used in this ComponentsDatabase, if any.
         """
         script = """
-document.addEventListener("DOMContentLoaded", function () {
+document.addEventListener("DOMContentLoaded", function () {{
     'use strict';
     var components,
-        baseUserEmbed = {'profilePicture': {},
-                         'emailAddresses': {} },
-        baseJobEmbed = {},
-        baseDomainEmbed = {logo: {},
-                           company: {},
-                           menus: {menuItems: {}}};
-    var components;
+        baseUserEmbed = {0},
+        baseJobEmbed = {1},
+        baseDomainEmbed = {2},
```

Comment:
I like this idea better than mine but when we start adding more entities to this list the arguments for generate_react_componenets might start to get out of control?

Reply:
It's already incredibly ugly. some kind of refactor is in order regardless